src/v2/models/AppState.js

```diff
@@ -46,9 +40,41 @@ export default Model.extend({
     },
   },
 
-  parse: function (options) {
-    this.settings = options.settings;
-    return _.extend(_.omit(options, 'settings'), { languageCode: this.settings.get('languageCode') });
+  getCurrentViewState () {
+    const currentFormName = this.get('currentFormName');
+    let vo;
+    if (!_.isEmpty(this.get('remediation'))) {
+      vo = this.get('remediation').filter(r => r.name === currentFormName)[0];
+    }
+
+    if (!vo) {
+      if (currentFormName) {
+        Logger.warn(`Cannot find view state for form ${currentFormName}. Fall back to terminal state.`);
+      }
+      vo = this.get('currentState').terminal[0];
```

Comment:
Can we add some comments here for our assumptions that when we don't have a remediation, we are in a terminal state ?